### PR TITLE
Formats inventory page with filter and items

### LINKED ISSUE #86: Format inventory page

### PR DESCRIPTION
**Pull request by lisa-sa-li**

### CODE REVIEW

File: hackathon_site/dashboard/frontend/src/components/inventory/InventoryFilter/InventoryFilter.js [hunk 1]
```diff
@@ -12,79 +12,74 @@ import FormControlLabel from "@material-ui/core/FormControlLabel";
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/components/inventory/InventoryFilter/InventoryFilter.js:11]: I can't comment on the line itself, but line 3 the `Paper` import can be removed

File: hackathon_site/dashboard/frontend/src/components/inventory/InventoryFilter/InventoryFilter.js [hunk 2]
```diff
@@ -12,79 +12,74 @@ import FormControlLabel from "@material-ui/core/FormControlLabel";
...
+                            label={item.name}
+                            control={<Checkbox color="primary" />}
+                        />
+                        <Chip label={item.qty} className={styles.filterCategoryChip} />
+                    </div>
+                ))}
+            </fieldset>
+        </form>
         <div className={styles.filterBtns}>
             <Button
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/components/inventory/InventoryFilter/InventoryFilter.js:83]: Once again I can't comment on the actual line, but should `fullWidth={true}` be set on the reset button? It's noticeable when being clicked with the animation

File: hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.js [hunk 3]
```diff
@@ -1,26 +1,151 @@
...
-                categories={inventoryCategories}
-                applyFilter={applyFilter}
-                removeFilter={removeFilter}
-            />
-        </div>
+            <div className={styles.inventory}>
+                <Drawer
+                    className={styles.inventoryFilterDrawer}
+                    variant="permanent"
+                    variant={mobileWidth ? "permanent" : "temporary"}
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.js:40]: Two variant attributes!

File: hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.js [hunk 4]
```diff
@@ -1,26 +1,151 @@
...
 const removeFilter = () => alert("Removes all filters and resets form");
 
 const Inventory = () => {
+    const [mobileOpen, setMobileOpen] = React.useState(false);
+
+    const toggleFilter = () => {
+        setMobileOpen(!mobileOpen);
+    };
+
+    const mobileWidth = useMediaQuery(useTheme().breakpoints.up("md"));
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.js:32]: This is cool

File: hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.js [hunk 5]
```diff
@@ -1,26 +1,151 @@
...
+                        categories={inventoryCategories}
+                        applyFilter={applyFilter}
+                        removeFilter={removeFilter}
+                    />
+                </Drawer>
+                <div className={styles.inventoryBody}>
+                    <Typography variant="h1">Hardware Inventory</Typography>
+
+                    <div className={styles.inventoryBodyToolbar}>
+                        <Grid
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.js]: How come you're using a grid container here? Swapping this out for a div seems to work fine for me, all the flexbox magic is maintained

File: hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.js [hunk 6]
```diff
@@ -1,26 +1,151 @@
...
+                                    aria-label="Orders"
+                                    startIcon={<FilterListIcon />}
+                                    onClick={toggleFilter}
+                                >
+                                    Filter
+                                </Button>
+                            </Hidden>
+
+                            <div className={styles.inventoryBodyToolbarRefresh}>
+                                <Typography variant="body2">123 items.</Typography>
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.js]: The most minor thing ever, but the fullstop feels weird. Can change this around when the page gets connected

File: hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.js [hunk 7]
```diff
@@ -1,26 +1,151 @@
...
+                            </IconButton>
+                        </Grid>
+
+                        <Divider
+                            orientation="vertical"
+                            className={styles.inventoryBodyToolbarDivider}
+                            flexItem
+                        />
+
+                        <Grid
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.js]: Ditto about grid container

**Review by grahamhoyes — COMMENTED** [review]
Functionality looks good for now for the most part.

There's going to be a bit of refactoring required when this gets hooked up to redux. The `Inventory` component will likely be renamed to `UnconnectedInventory`, and take in props for search results, total quantity, etc. Then the `ConnectedInventory` will be hooked up to the redux store and exported. I've created an issue to capture that work: https://github.com/ieeeuoft/hackathon-template/issues/128

File: hackathon_site/dashboard/frontend/src/components/inventory/InventoryFilter/InventoryFilter.js (hunk 2 above)
Comment (lisa-sa-li) [reply hackathon_site/dashboard/frontend/src/components/inventory/InventoryFilter/InventoryFilter.js:83]: It was meant to be smol

**Review by grahamhoyes — APPROVED** [review]

**Pull request merged**